changelog/action.yml

```diff
@@ -65,13 +65,29 @@ runs:
         t=$(mktemp)
         printf "# Changelog\n\n" | cat - CHANGELOG.md > "$t" && mv "$t" CHANGELOG.md
 
-        git stash || true
-        git checkout master || true
-        git stash pop || true
-        git add CHANGELOG.md
-        git commit -m "autogen(docs): regenerate and update changelog" || true
-        git pull origin master --rebase || true
-        if [[ "$GITHUB_REF_TYPE" == "branch" ]]; then
-          git push origin "$GITHUB_REF_NAME" || true
+
+        if [[ "$GITHUB_REF_TYPE" != "tag" ]]; then
+          if [[ "$(git log --format=%B -n1)" =~ "autogen"* ]]; then
+            echo "No-op because the previous commit was autogen"
```

Comment:
To avoid a lot of nesting, would it make sense to just `exit 0`? Also I wam wondering whether we can use github's "if" statements to not run this at all: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idif